torchvision/prototype/datasets/_builtin/stanford_cars.py

```diff
@@ -0,0 +1,97 @@
+from typing import Any, Dict, List, Tuple, Iterator
+
+import numpy as np
+from torchdata.datapipes.iter import Filter, IterDataPipe, Mapper, Zipper
+from torchvision.prototype.datasets.utils import Dataset, DatasetConfig, DatasetInfo, HttpResource
+from torchvision.prototype.datasets.utils._internal import hint_sharding, hint_shuffling, path_comparator, read_mat
+from torchvision.prototype.features import BoundingBox, EncodedImage
+
+
+class _StanfordCarsLabelReader(IterDataPipe[Tuple[np.ndarray, int]]):
```

Comment:
1. You can make this "public" since this part of `torchvision` is private as a whole and we only explicitly expose stuff.
2. The generic type annotation has to match the one from `__iter__`. See the comment below for an explanation
3. 
```suggestion
class StanfordCarsLabelReader(IterDataPipe[Tuple[int, int, int, int, int, str]]):
```